src/soc/ibm/power9/istep_8_6.c

```diff
@@ -0,0 +1,280 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <stdint.h>
+#include <console/console.h>
+#include <soc/istep8.h>
+
+/* From src/import/chips/p9/procedures/hwp/nest/p9_fbc_eff_config.C */
+/* LE epsilon (2 chips per-group) */
+static const uint32_t EPSILON_R_T0_LE[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T1_LE[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T2_LE[] = {   67,   69,   71,   74,   79,  103 };
+static const uint32_t EPSILON_W_T0_LE[] = {    0,    0,    0,    0,    0,    5 };
+static const uint32_t EPSILON_W_T1_LE[] = {   15,   16,   17,   19,   21,   33 };
+
+/* HE epsilon (4 chips per-group) */
+static const uint32_t EPSILON_R_T0_HE[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T1_HE[] = {   92,   94,   96,   99,  104,  128 };
+static const uint32_t EPSILON_R_T2_HE[] = {  209,  210,  213,  216,  221,  245 };
+static const uint32_t EPSILON_W_T0_HE[] = {   28,   29,   30,   32,   34,   46 };
+static const uint32_t EPSILON_W_T1_HE[] = {  117,  118,  119,  121,  123,  135 };
+
+/* HE epsilon (flat 4 Zeppelin) */
+static const uint32_t EPSILON_R_T0_F4[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T1_F4[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T2_F4[] = {   83,   84,   87,   90,   95,  119 };
+static const uint32_t EPSILON_W_T0_F4[] = {    0,    0,    0,    0,    0,    5 };
+static const uint32_t EPSILON_W_T1_F4[] = {   18,   19,   20,   22,   24,   36 };
+
+/* HE epsilon (flat 8 configuration) */
+static const uint32_t EPSILON_R_T0_F8[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T1_F8[] = {    7,    7,    8,    8,   10,   22 };
+static const uint32_t EPSILON_R_T2_F8[] = {  148,  149,  152,  155,  160,  184 };
+static const uint32_t EPSILON_W_T0_F8[] = {    0,    0,    0,    0,    0,    5 };
+static const uint32_t EPSILON_W_T1_F8[] = {   76,   77,   78,   80,   82,   94 };
+
+
+/* Keep them global and later we can use function to retrieve the values set here */
+
+uint32_t eps_r[NUM_EPSILON_READ_TIERS];
+uint32_t eps_w[NUM_EPSILON_WRITE_TIERS];
+int8_t eps_guardband;
+uint8_t core_floor_ratio;
+uint8_t core_ceiling_ratio;
+uint32_t freq_fbc;
+uint32_t freq_core_ceiling;
+
+static int p9_calculate_frequencies(uint8_t *floor_ratio, uint8_t *ceiling_ratio,
+				    uint32_t *fbc_freq, uint32_t *freq_ceiling)
+{
+	/* Default 4800 from src/usr/targeting/common/xmltohb/hb_customized_attrs.xml */
+	uint32_t freq_core_floor = 4800; 
+	/* According to src/usr/targeting/common/xmltohb/attribute_types.xml
+	   ATTR_FREQ_CORE_NOMINAL_MHZ maps to NOMINAL_FREQ_MHZ present in talos.xml */
+	uint32_t freq_core_nom = 4800;
+	uint8_t async_safe_mode = FABRIC_ASYNC_PERFORMANCE_MODE;
+	*freq_ceiling = 4800;
```

Comment:
https://github.com/3mdeb/openpower-coreboot-docs/blob/main/logs/scom_dumps/istep_6_12_host_voltage_config.log#L40 and surrounding lines. Obviously the default values are not the same as actual.